### PR TITLE
[IMP] open_academy: Added wizard model that enables users to register multiple attendees to multiple sessions at the same time; added spanish translation to module T#53764

### PR DESCRIPTION
**Pull request by payen000**
On the session tree view, selecting sessions enables the 'register attendees' button on the action button:
image

Clicking on this button displays the wizard, from which multiple attendees can be selected for registration:
image

The wizard raises an error when trying to register more users than there are available seats:
image

If the registration is successful, the sessions are updated to reflect this:
image

Finally, a spanish translation of the module is available:
image

image

### CODE REVIEW

**Comment by payen000** [comment]
@frahikLV Hello again, would you review my code please? :)

File: open_academy/wizard/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from . import wizard_session
```
Comment (frahikLV) [inline open_academy/wizard/__init__.py:1]: change all the references of the name of the wizard from `wizard_session` to `assign_atendee_sessions`

File: open_academy/wizard/wizard_session.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+
+class WizardSession(models.TransientModel):
```
Comment (frahikLV) [inline open_academy/wizard/wizard_session.py:5]: rename it

File: open_academy/wizard/wizard_session.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    _name = 'wizard_session'
```
Comment (frahikLV) [inline open_academy/wizard/wizard_session.py:6]: modes `_name` don't use a underscore in the name, uses `.` , change this when you rename the wizard

File: open_academy/wizard/wizard_session.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    _description = 'Wizard Session'
```
Comment (frahikLV) [inline open_academy/wizard/wizard_session.py:7]: rename

File: open_academy/wizard/wizard_session.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    session_ids = fields.Many2many('session', required=True, default=lambda self: self._default_session_ids())
+    attendee_ids = fields.Many2many('res.partner')
+
+    @api.model
+    def _default_session_ids(self):
+        selected_session_ids = self.env['session'].browse(self._context.get('active_ids'))
+        return selected_session_ids
```
Comment (frahikLV) [inline open_academy/wizard/wizard_session.py:15]: Remove this, please send the ids on the action call the ids via context with `default_session_ids` key

File: open_academy/wizard/wizard_session.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+
+class WizardSession(models.TransientModel):
+    _name = 'wizard_session'
+    _description = 'Wizard Session'
+
+    session_ids = fields.Many2many('session', required=True, default=lambda self: self._default_session_ids())
```
Comment (frahikLV) [inline open_academy/wizard/wizard_session.py:9]:
```suggestion
    session_ids = fields.Many2many('session', required=True)
```

File: open_academy/wizard/wizard_session.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    attendee_ids = fields.Many2many('res.partner')
+
+    @api.model
+    def _default_session_ids(self):
+        selected_session_ids = self.env['session'].browse(self._context.get('active_ids'))
+        return selected_session_ids
+
+    @api.depends('session_ids', 'attendee_ids')
```
Comment (frahikLV) [inline open_academy/wizard/wizard_session.py:17]: remove this

File: open_academy/wizard/wizard_session.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+            session.attendee_ids |= self.attendee_ids
+
+            if session.number_of_seats < len(session.attendee_ids):
+                session.attendee_ids = old_attendee_ids
+                overregistered_sessions.append(session.name)
+
+        if overregistered_sessions:
+            overregistered_sessions = ', '.join(overregistered_sessions)
+            error_message = f'Aborted registration in {overregistered_sessions}: attendee limit exceeded.'
+            raise ValidationError(_(error_message))
```
Comment (frahikLV) [inline open_academy/wizard/wizard_session.py:32]: Refactor this code, please don't made a lot of talks to DB to update the `attendee_ids`, first check if the session has enough space, if not, show a UserError, not a ValidationError
If the sessions have space, made a `write` to all the records including with the `Command.link()` method

**Review by frahikLV — CHANGES_REQUESTED** [review]
@payen000 can you address these changes?

**Comment by frahikLV** [comment]
@payen000 also fix your translation file... has a lot of lines, only translate the records related to your module

**Comment by payen000** [comment]
@frahikLV Sorry I couldn't finish sooner, I got a little stuck in a certain part... Would you review my code? If not, I understand, I will wait until monday with Germana :)

**Comment by payen000** [comment]
@goliveirab Hello, would you review my code please? :)

**Pull request merged**